### ISSUE TITLE
Impact of closure of user-agent-string.info project?

### ISSUE THREAD

**Issue by jugglingcats** [arouel/uadetector#93]
Hi, there is a message at http://user-agent-string.info saying:

```
user-agent-string project will end on November 30, 2014.
The new version can be found at udger.com
```

That's in six days time. It doesn't look like the new site is ready for prime time yet though.

Am actually happy to help fund user-agent-string.info service and the published price seems reasonable. I just wonder if there is going to be any impact on uadetector as this is what we've integrated with and am very happy with it!

**Comment by avila-devlogic** [comment]
Hi @before 

I'm interested in creation and maintenance of browscap.ini support.
I'm aware that you've already took a look at it and you were not satisfied with it, but that's the best bet we can get from open source community and by providing unified Java library for it we will expand it's use and contributions to it.

What I want to know is if this library will stay alive for me to contribute to ?
It came far away and I don't want all those people using it run away from it due to subscriptions or inability for auto-update.

Thanks in advance,
Ahmed

**Comment by HaraldWalker** [comment]
Any news on this? If udger.com offers the datafile in a user-agent-string.info compatible format for registered users (in a free and paid version), isn't the change then just to replace the hardcoded download url with a user configurable url?

**Comment by before** [comment]
@avila-devlogic yes, if you are interested I open the repository for you. Than you can develop directly in a topic branch and I would also contribute to this. I'm not very motivated to develop this for my own. Any contribution is welcome.

@HaraldWalker What do you suggest? Which database is out there that could bundled with our library?

**Comment by tjacoby** [comment]
For all customers of udger.com it would be nice to easily switch the url to the private udger url. This can easily be done with the current code. But parsing the XML file will fail because the path to the DTD was changed in the udger.com files. This leads to an exception in the class XmlDataHandler and the default DTD path cannot be (easily) overridden without duplicating a lot of code to instantiate our own version of the XmlDataHandler:

See:
```
protected static final String UASDATA_DEF_URL = "http://user-agent-string.info/rpc/uasxmldata.dtd";
```

```
net.sf.uadetector.internal.data.XmlDataHandler.resolveEntity(String, String)
```

We could open a pull request to allow overriding the DTD location which would make this great library usable again.

**Comment by before** [comment]
@tjacoby good job. I like your changes. Only the indentation is different to mine.

**Comment by ESPNMichaelKidd** [comment]
@tjacoby @before I'm struggling to figure out how to reconfigure uadetector to use the data from udger.com.   Would someone mind helping me figure it out with some sample code?

**Comment by HaraldWalker** [comment]
@ESPNMichaelKidd There hasn't been a new release yet but if you don't mind building the jar yourself (and using a local maven repository), then you can use the commercial udger.com database. 
The UADetectorServiceFactory can now be initialized using your own data and version URLs (so you use the new udger.com link including the access key you get for a paid subscription).

**Comment by ESPNMichaelKidd** [comment]
@HaraldWalker many thanks, I see how it works now.
